Makefile

```diff
@@ -112,6 +114,17 @@ ifeq ($(ENABLE_KVM), 1)
 CARGO_OSDK_ARGS += --qemu-args="-accel kvm"
 endif
 
+ifneq ($(filter fetch, $(MAKECMDGOALS)),)
+undefine OFFLINE
+# Use a nonexistent test name to skip all tests, as the goal is to cache dependencies during build.
+CARGO_OSDK_TESTNAME := nonexistent-test
+endif
```

Comment:
I don't really get the point. It seems that if not running `make fetch`(`make ktest` for example), the `--offline` argument will not be used.

Could you please elaborate?

Reply:
The main issue is that there are many conditional dependencies across different targets, and we want `--offline` to always work in all scenarios, such as building different architectures, building after a clean, or running `make ktest`.

If `make fetch` is not run beforehand, the dependency is incomplete. For instance:

- Running `make build` only fetches dependencies for that specific target, so trying `make ktest` afterward might fail due to missing dependencies.
- After `make clean`, OSDK regenerates the `base` crate and `cargo build` will try to connect to the network to check for unnecessary dependencies (e.g., RISCV-related ones), which won't be cached. This leads to `--offline` to fail, even if all X86 dependencies are cached.

Reply:
OK, so this is just a workaround. And I assume in this PR offline cargo osdk test won't work.

So would a proper solution be supporting `cargo osdk fetch test`, which fetches dependencies specific to ktest?

Reply:
make fetch also caches ktest dependencies. I didn't implement cargo osdk fetch test because we don't know which directories to cd into to run it. We could add a function that simply forwards args to cargo osdk test.
Sorry that my bandwidth is very limited right now, so if further changes are needed, I might be slow to update.